packages/@glimmer/runtime/lib/environment.ts

```diff
@@ -44,137 +39,19 @@ import {
 import { assert, WILL_DROP, DID_DROP, expect, symbol } from '@glimmer/util';
 import { AttrNamespace, SimpleElement } from '@simple-dom/interface';
 import { DOMChangesImpl, DOMTreeConstruction } from './dom/helper';
-import { ConditionalReference, UNDEFINED_REFERENCE } from './references';
+import { ConditionalReference } from './references';
 import { DynamicAttribute, dynamicAttribute } from './vm/attributes/dynamic';
 import { RuntimeProgramImpl } from '@glimmer/program';
-
-export function isScopeReference(s: ScopeSlot): s is VersionedPathReference {
-  if (s === null || Array.isArray(s)) return false;
-  return true;
-}
-
-export class ScopeImpl<C extends JitOrAotBlock> implements PartialScope<C> {
```

Comment:
This was moved to `scope.ts` since the environment.ts file was getting large and covering too many concerns.

Reply:
Will break this out into a separate PR